Makefile

```diff
@@ -0,0 +1,57 @@
+PROJECT := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+SRC := $(PROJECT)src
+TESTS := $(PROJECT)tests
+ALL := $(SRC) $(TESTS)
+
+export PYTHONPATH = $(PROJECT):$(PROJECT)/lib:$(SRC)
+
+update-dependencies:
+	uv lock -U --no-cache
+
+generate-requirements:
+	uv pip compile -q --no-cache pyproject.toml -o requirements.txt
+	uv pip compile -q --no-cache --all-extras pyproject.toml -o requirements-dev.txt
+
+lint:
+	uv tool run ruff check $(ALL)
+	uv tool run ruff format --check --diff $(ALL)
+
+fmt:
+	uv tool run ruff check --fix-only $(ALL)
+	uv tool run ruff format $(ALL)
+
+unit:
+	uv run --all-extras \
+		coverage run \
+		--source=$(SRC) \
+		-m pytest \
+		--ignore=$(TESTS)/integration \
```

Comment:
I prefer pointing at the test folders directly (`./tests/unit`, `./tests/scenario`) rather than pointing at the root and then `--ignore`ing things, but both work

Reply:
yeah that's a good point, I like that more too

Reply:
however it gets more difficult to work with `$(ARGS)` in that case, because you can't `make unit ./just/this/test.py` anymore, which I think is the reason for the --exclude strategy

Reply:
oh yeah I see what you mean.  

hmm I can't think of a nicer way.  Personally I prefer removing `--ignore` but that's mostly because I never do `make unit ./just/this/test.py`.  But I can see if someone uses it, losing that sucks